### PR TITLE
Air Sensor are Craftable, Removed from RPD UI, Have New Linking System

### PR DESCRIPTION
**Pull request by SyncIt21**
## About The Pull Request

**1. Craftable & Removed from RPD UI**

1. Air sensor's are now craftable
2. You can turn them on/off with hand. Even though turning off the sensor will change it to a diffrent type[from `obj/machinery/air_sensor` -> `obj/item/air_sensor`] it's I/O port's are sill preserved when turning them on although you have to assign it a new name again which is usefull if you want to change the sensor's purpose in game.
3. They can now only be deconstructed by a welding tool and should be wrenched in place to turn them on. 
4. Turned off air sensor's once unwrenched can be picked up like any regular item 
5. Air sensor's are removed from the RPD UI because they don't go with pipes so it logically doesn't make sense to group them with pipe related device's

Removed unused code in the process

https://github.com/tgstation/tgstation/assets/110812394/3439a0f3-9c48-43ac-8f4b-98135435ec13

**2. New ID System**
The problem with air sensor's is that each sensor is assigned a unique ID which is then stored in `GLOB.objects_by_id_tag` list. Each sensor name it's assigned based on the gas it's trying to sense(for naming only even though it can detect other gases) So if 2 sensor's having the same ID are made they will overwrite each other in this list leaving one sensor orphaned in the world which cannot be referenced because it's value was overwritten by a new sensor having the same ID in this list.

The Solution? Rather than having all atmos computer's look up sensor's from this 1 global list make each computer keep track of all sensor's it's responsible for in it's own local list[which i called `connected_sensor's`] this way 2 sensor's can have randomly generated names in the global `GLOB.objects_by_id_tag` list but the computer will know what sensor to look up in this list based on the stored sensor ID's in the  `connected_sensor's` list

Basically what i am getting at is now you can make as many air sensor's as you wish but you will know have to connect that sensor to the computer using a multitool. 
Notice in the video how i made 2 sensor's called `Supermatter Chamber Sensor's`] and every time you try to connect an sensor which has the same name[`Supermatter Chamber Sensor's` in this case] they will ovewrite the old sensor in it's list as shown in the video

https://github.com/tgstation/tgstation/assets/110812394/b5283c3b-c8a1-4b94-a6a8-8ba7a0007615

**Why it's good for the game**
Screenshot (247)

I agree. Also air sensor's taking up a full Tab/Section in the RPD UI wasted a lot of UI space so that's removed now. Also making the air sensor's wrenchable and pickable item's was also requested in https://github.com/tgstation/tgstation/pull/72019#issuecomment-1355499873 so you relate them to device's like meter's

Another huge issue was that the number of air sensor's you can make in the world was limited because each sensor in the world must have a unique ID but that's finally fixed now so yeah make as many sensor's as you want. 

## Changelog
:cl:
add: air sensor's are craftable
refactor: air sensor's can now be turned off by hand and can only be deconstructed by a welding tool
refactor: removed `Params()` proc
qol: unwrenched air sensors can be picked up & recycled like regular item's
del: air sensor are removed from the RPD UI
qol: air sensor's are no longer restricted by their unique ID's which mean you can craft as many air sensors as you want.
/:cl:

### CODE REVIEW

**Comment by MTandi** [comment]
> Also the same question could be asked for every component in the atmospheric's crafting section like why can we craft meter's but also spawn them with the RPED? There is really no harm in giving the player more option's so it's ok to look past this

I just want to point out that meter is not an independent machine. It works only with pipe. While air sensors and burn chambers are unrelated to pipes. That's my train of though when I said that I think it shouldn't be in Rapid PIPE Dispenser. But I'm OK with whatever you feel right.

**Comment by SyncIt21** [comment]
`I just want to point out that meter is not an independent machine. It works only with pipe. While air sensors and igniters are unrelated to pipes.`

You know what you changed my mind that does make a lot of sense. Air sensor's are meant to be independent from everything so yeah sure i'll remove it from the RPD

**Comment by TheVekter** [comment]
> You can turn them on/off with hand. Turning off an air sensor will also disconnect it's linked I/O port's so you will have to reconnect them again

Is there any particular reason you need to reconnect the I/O port? I like the idea of being able to switch them off if they're being annoying over nothing, but I don't like having to re-link them to turn them back on.

File: code/game/machinery/computer/atmos_computers/_air_sensor.dm [hunk 1]
```diff
@@ -72,8 +85,110 @@
...
 		balloon_alert(user, "connected to output")
-	else
-		multi_tool.buffer = src
-		balloon_alert(user, "added to multitool buffer")
 
 	return TRUE
+
+/obj/item/air_sensor
+	name = "Air Sensor"
+	desc = "It's an switched off air sensor."
```
Comment (TheVekter) [inline code/game/machinery/computer/atmos_computers/_air_sensor.dm:93]: "It's a switched-off air sensor." I'm not sure about this for a description, actually. Maybe something like "A device designed to detect gases and their concentration in an area." We need to describe what an air sensor is to players who maybe don't know.

File: code/datums/components/crafting/atmospheric.dm [hunk 2]
```diff
@@ -17,6 +17,18 @@
 	time = 0.5 SECONDS
 	category = CAT_ATMOSPHERIC
 
+/datum/crafting_recipe/air_sensor
+	name = "Monitered Air Sensor"
```
[the remark this answers is not in the document]
Comment (SyncIt21) [reply code/datums/components/crafting/atmospheric.dm:21]: Yup

File: code/game/machinery/computer/atmos_computers/_air_sensor.dm (hunk 1 above)
Comment (SyncIt21) [reply code/game/machinery/computer/atmos_computers/_air_sensor.dm:93]: Yeah i have no clue how to be more descriptive, i'll do that

File: code/datums/components/crafting/atmospheric.dm (hunk 2 above)
[the remark this answers is not in the document]
Comment (TheVekter) [reply code/datums/components/crafting/atmospheric.dm:21]: All good, I just like to do spelling/grammar passes when I can.

**Comment by MTandi** [comment]
> I'm still annoyed at the fact that air sensor's have unique ID's attached to them so even if they are crafted they can't be turned on if the selected ID from the input list is taken. I'll undraft this when i find a solution to that. Tomorrow that is

Maybe use controllers like blast doors and buttons do. They have roundstart text IDs but you can't input anything but numbers though.

**Comment by SyncIt21** [comment]
> > I'm still annoyed at the fact that air sensor's have unique ID's attached to them so even if they are crafted they can't be turned on if the selected ID from the input list is taken. I'll undraft this when i find a solution to that. Tomorrow that is
> 
> Maybe use controllers like blast doors and buttons do. They have roundstart text IDs but you can't input anything but numbers though.

Multiple Blast door controller's can share the same ID which will allow multiple button's to control multiple door's. There is no problem here because it's an M:N relation(M=multiple controller's, N= multiple door's)

Air sensors can't use this model because only 1 air sensor should be controlled by 1 atmospheric station monitoring console i.e it's a 1:1 relation. So if i do create a controller for air sensors and allow multiple controller's to have the same ID then the  atmospheric station monitoring console will be unable to differentiate between 2 sensor's having the same ID.

The solution is for each computer to have their own local list of sensors they control and not one global list `GLOB.objects_by_id` where one 2 object's with the same ID can overwrite each other on this list.

I have some idea's, I'll look into it tomorrow cause it's 12 pm midnight here. gn

**Comment by SyncIt21** [comment]
Yes finally solved no more unique ID problem, also even better news the I/O port's will be preserved when turning the sensor on/off. Read updated PR body

**Comment by SyncIt21** [comment]
I have no pending work, need someone to give this some attention.
@LemonInTheDark merge/review?

**Comment by SyncIt21** [comment]
I'm getting desperate, I really don't want this to go stale. @Time-Green  nothing in this PR particularly concerns you but still maybe you could merge/review this :) ?
I'm not on the discord so i don't know other maintainer names

File: code/game/machinery/computer/atmos_computers/_air_sensor.dm [hunk 3]
```diff
@@ -27,21 +39,45 @@
 	reset()
 	return ..()
 
+/obj/machinery/air_sensor/return_air()
+	if(!on)
+		return null
```
Comment (Time-Green) [inline code/game/machinery/computer/atmos_computers/_air_sensor.dm:44]:
```suggestion
		return 
```

File: code/game/machinery/computer/atmos_computers/_air_sensor.dm [hunk 4]
```diff
@@ -69,11 +103,121 @@
...
+		multi_tool.buffer = null
 		balloon_alert(user, "connected to output")
+
 	else
 		multi_tool.buffer = src
 		balloon_alert(user, "added to multitool buffer")
 
 	return TRUE
+
+/obj/item/air_sensor
```
Comment (Time-Green) [inline code/game/machinery/computer/atmos_computers/_air_sensor.dm:117]: Can you autodoc the type?

File: code/game/machinery/computer/atmos_computers/_air_sensor.dm [hunk 5]
```diff
@@ -69,11 +103,121 @@
...
+	if(default_unfasten_wrench(user, tool) == SUCCESSFUL_UNFASTEN)
+		return TOOL_ACT_TOOLTYPE_SUCCESS
+	return
+
+/obj/item/air_sensor/welder_act(mob/living/user, obj/item/tool)
+	if(!tool.tool_start_check(user, amount = 1))
+		return
+
+	loc.balloon_alert(user, "dismantling sensor")
+	if(!tool.use_tool(src, user, 20, volume = 30, amount = 1))
```
Comment (Time-Green) [inline code/game/machinery/computer/atmos_computers/_air_sensor.dm:212]:
```suggestion
	if(!tool.use_tool(src, user, 2 SECONDS, volume = 30, amount = 1))
```

Please use the time defines

File: code/game/machinery/computer/atmos_computers/inlets.dm [hunk 6]
```diff
@@ -1,6 +1,7 @@
 /obj/machinery/atmospherics/components/unary/outlet_injector/monitored
 	on = TRUE
 	volume_rate = MAX_TRANSFER_RATE
+	var/chamber_id
```
Comment (Time-Green) [inline code/game/machinery/computer/atmos_computers/inlets.dm:4]: Autodoc :)

**Review by Time-Green — CHANGES_REQUESTED** [review]
Looks good! Minor nitpicks. Sorry this got ignored so hard, just happens sometimes, you did right by reaching out

**Pull request merged**